org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/StringTemplateComponent.java

```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.core.dom;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.internal.core.dom.util.DOMASTUtil;
+
+/**
+ * StringTemplateComponent  AST node type.
+ *
+ * @since 3.36
+ * @noreference This class is not intended to be referenced by clients.
+ */
+@SuppressWarnings("rawtypes")
+public class StringTemplateComponent extends Expression {
+	/**
+	 * A list of property descriptors (element type:
+	 * {@link StructuralPropertyDescriptor}),
+	 * or null if uninitialized.
+	 */
+	private static final List PROPERTY_DESCRIPTORS;
+
+	/**
+	 * The "start" structural property of this node type (child type: {@link StringFragment}).
+	 */
+	public static final ChildPropertyDescriptor EMBEDDED_EXPRESSION_PROPERTY =
+		new ChildPropertyDescriptor(StringTemplateComponent.class, "expression", Expression.class, MANDATORY, CYCLE_RISK); //$NON-NLS-1$
+
+	/**
+	 * The "start" structural property of this node type (child type: {@link StringFragment}).
+	 */
+	public static final ChildPropertyDescriptor STRING_FRAGMENT_PROPERTY =
+		new ChildPropertyDescriptor(StringTemplateComponent.class, "fragment", StringFragment.class, MANDATORY, CYCLE_RISK); //$NON-NLS-1$
+
+	static {
+		List propertyList = new ArrayList(2);
```

Comment:
This needs to b 3 - as per convention - and also to optimize [ we have no_of_props+1 nodes at the minimum - so an initial list of 3 would mean no reallocation [not going into details of library implementation]